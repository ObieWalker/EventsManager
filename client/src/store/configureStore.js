import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import rootReducer from '../reducers/rootReducer';

const middleware = applyMiddleware(thunkMiddleware, createLogger({ collapsed: true }));

const Store = createStore(
    rootReducer,
    applyMiddleware(middleware)
);

export default Store;