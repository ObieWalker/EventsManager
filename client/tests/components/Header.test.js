import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import connectHeader, { Header } from '../../src/components/Header';

let mountedLogout;
let props;
const locations = [];
const history = {
  push(location) {
    locations.push(location);
  }
};

const login = { isAuthenticated: false, user: {} };
const logOut = jest.fn(() => Promise.resolve());
const mockStore = configureStore();
let wrapper;

const getComponent = () => {
  if (!mountedLogout) {
    props = {
      login,
      logOut,
      history
    };
    history.push = jest.fn();
    mountedLogout = shallow(<Header {...props} />);
  }
  return mountedLogout;
};

describe('Logout Component', () => {
  beforeEach(() => {});

  it('renders component successfully', () => {
    wrapper = getComponent();
    expect(wrapper).toMatchSnapshot();
  });

  it('testing component will recieve props', () => {
    const nextProps = {
      login: {
        isAuthenticated: true
      }
    };
    wrapper = getComponent();
    wrapper.instance().componentWillReceiveProps(nextProps);
    expect(wrapper.state().loggedIn).toEqual(true);
  });

  it('renders component successfully', () => {
    wrapper = getComponent();
    expect(wrapper.find('Link').length).toBe(5);
    expect(wrapper.find('ul').length).toBe(1);
    expect(wrapper.find('div').length).toBe(3);
    expect(wrapper.find('nav').length).toBe(1);
    expect(wrapper.find('button').length).toBe(2);
  });

  it('testing user login function', () => {
    wrapper = getComponent();
    const e = { preventDefault: () => undefined };
    e.preventDefault = jest.fn();
    wrapper.instance().handleLogOut(e);
    expect(e.preventDefault).toHaveBeenCalled();
  });

  it('testing user login function', () => {
    wrapper = getComponent();
    const e = { preventDefault: () => undefined };
    e.preventDefault = jest.fn();
    wrapper.instance().handleLogOut(e);
    expect(history.push).toHaveBeenCalled();
  });
});
describe('Connected Component', () => {
  describe('Connected mountedLogout', () => {
    it('component successfully rendered', () => {
      const store = mockStore({
        login
      });
      wrapper = shallow(<connectHeader store={store} history={history} />);
      expect(wrapper.length).toBe(1);
    });
  });
});
