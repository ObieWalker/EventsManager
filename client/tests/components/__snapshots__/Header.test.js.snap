// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout Component renders component successfully 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    history={
      Object {
        "push": [MockFunction],
      }
    }
    logOut={[MockFunction]}
    login={
      Object {
        "isAuthenticated": false,
        "user": Object {},
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <nav
        className="indigo"
      >
        <div
          className="nav-wrapper "
        >
          <Link
            className="left"
            replace={false}
            style={
              Object {
                "color": "white",
                "fontSize": "30px",
                "paddingLeft": "10px",
                "textDecoration": "none",
              }
            }
            to="/"
          >
            The Events Manager
          </Link>
          <ul
            className="right hide-on-med-and-down"
            style={
              Object {
                "paddingRight": "20px",
              }
            }
          >
            <li>
              <Link
                replace={false}
                to="/"
              >
                Home
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/contact-us"
              >
                Contact Us
              </Link>
            </li>
            <div
              className="btn-group"
            >
              <li>
                <Link
                  replace={false}
                  to="/login"
                >
                  <button
                    className="btn-success btn-sm"
                    type="button"
                  >
                    Login
                  </button>
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/register"
                >
                  <button
                    className="btn-primary btn-sm"
                    type="button"
                  >
                    Register
                  </button>
                </Link>
              </li>
            </div>
          </ul>
        </div>
      </nav>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="nav-wrapper "
        >
          <Link
            className="left"
            replace={false}
            style={
              Object {
                "color": "white",
                "fontSize": "30px",
                "paddingLeft": "10px",
                "textDecoration": "none",
              }
            }
            to="/"
          >
            The Events Manager
          </Link>
          <ul
            className="right hide-on-med-and-down"
            style={
              Object {
                "paddingRight": "20px",
              }
            }
          >
            <li>
              <Link
                replace={false}
                to="/"
              >
                Home
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/contact-us"
              >
                Contact Us
              </Link>
            </li>
            <div
              className="btn-group"
            >
              <li>
                <Link
                  replace={false}
                  to="/login"
                >
                  <button
                    className="btn-success btn-sm"
                    type="button"
                  >
                    Login
                  </button>
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/register"
                >
                  <button
                    className="btn-primary btn-sm"
                    type="button"
                  >
                    Register
                  </button>
                </Link>
              </li>
            </div>
          </ul>
        </div>,
        "className": "indigo",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="left"
              replace={false}
              style={
                Object {
                  "color": "white",
                  "fontSize": "30px",
                  "paddingLeft": "10px",
                  "textDecoration": "none",
                }
              }
              to="/"
            >
              The Events Manager
            </Link>,
            <ul
              className="right hide-on-med-and-down"
              style={
                Object {
                  "paddingRight": "20px",
                }
              }
            >
              <li>
                <Link
                  replace={false}
                  to="/"
                >
                  Home
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/contact-us"
                >
                  Contact Us
                </Link>
              </li>
              <div
                className="btn-group"
              >
                <li>
                  <Link
                    replace={false}
                    to="/login"
                  >
                    <button
                      className="btn-success btn-sm"
                      type="button"
                    >
                      Login
                    </button>
                  </Link>
                </li>
                <li>
                  <Link
                    replace={false}
                    to="/register"
                  >
                    <button
                      className="btn-primary btn-sm"
                      type="button"
                    >
                      Register
                    </button>
                  </Link>
                </li>
              </div>
            </ul>,
          ],
          "className": "nav-wrapper ",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "The Events Manager",
              "className": "left",
              "replace": false,
              "style": Object {
                "color": "white",
                "fontSize": "30px",
                "paddingLeft": "10px",
                "textDecoration": "none",
              },
              "to": "/",
            },
            "ref": null,
            "rendered": "The Events Manager",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  <Link
                    replace={false}
                    to="/"
                  >
                    Home
                  </Link>
                </li>,
                <li>
                  <Link
                    replace={false}
                    to="/contact-us"
                  >
                    Contact Us
                  </Link>
                </li>,
                undefined,
                <div
                  className="btn-group"
                >
                  <li>
                    <Link
                      replace={false}
                      to="/login"
                    >
                      <button
                        className="btn-success btn-sm"
                        type="button"
                      >
                        Login
                      </button>
                    </Link>
                  </li>
                  <li>
                    <Link
                      replace={false}
                      to="/register"
                    >
                      <button
                        className="btn-primary btn-sm"
                        type="button"
                      >
                        Register
                      </button>
                    </Link>
                  </li>
                </div>,
              ],
              "className": "right hide-on-med-and-down",
              "style": Object {
                "paddingRight": "20px",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/"
                  >
                    Home
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Home",
                    "replace": false,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "Home",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/contact-us"
                  >
                    Contact Us
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Contact Us",
                    "replace": false,
                    "to": "/contact-us",
                  },
                  "ref": null,
                  "rendered": "Contact Us",
                  "type": [Function],
                },
                "type": "li",
              },
              undefined,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <li>
                      <Link
                        replace={false}
                        to="/login"
                      >
                        <button
                          className="btn-success btn-sm"
                          type="button"
                        >
                          Login
                        </button>
                      </Link>
                    </li>,
                    <li>
                      <Link
                        replace={false}
                        to="/register"
                      >
                        <button
                          className="btn-primary btn-sm"
                          type="button"
                        >
                          Register
                        </button>
                      </Link>
                    </li>,
                  ],
                  "className": "btn-group",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Link
                        replace={false}
                        to="/login"
                      >
                        <button
                          className="btn-success btn-sm"
                          type="button"
                        >
                          Login
                        </button>
                      </Link>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": <button
                          className="btn-success btn-sm"
                          type="button"
                        >
                          Login
                        </button>,
                        "replace": false,
                        "to": "/login",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Login",
                          "className": "btn-success btn-sm",
                          "type": "button",
                        },
                        "ref": null,
                        "rendered": "Login",
                        "type": "button",
                      },
                      "type": [Function],
                    },
                    "type": "li",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Link
                        replace={false}
                        to="/register"
                      >
                        <button
                          className="btn-primary btn-sm"
                          type="button"
                        >
                          Register
                        </button>
                      </Link>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": <button
                          className="btn-primary btn-sm"
                          type="button"
                        >
                          Register
                        </button>,
                        "replace": false,
                        "to": "/register",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Register",
                          "className": "btn-primary btn-sm",
                          "type": "button",
                        },
                        "ref": null,
                        "rendered": "Register",
                        "type": "button",
                      },
                      "type": [Function],
                    },
                    "type": "li",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
      "type": "nav",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav
          className="indigo"
        >
          <div
            className="nav-wrapper "
          >
            <Link
              className="left"
              replace={false}
              style={
                Object {
                  "color": "white",
                  "fontSize": "30px",
                  "paddingLeft": "10px",
                  "textDecoration": "none",
                }
              }
              to="/"
            >
              The Events Manager
            </Link>
            <ul
              className="right hide-on-med-and-down"
              style={
                Object {
                  "paddingRight": "20px",
                }
              }
            >
              <li>
                <Link
                  replace={false}
                  to="/"
                >
                  Home
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/contact-us"
                >
                  Contact Us
                </Link>
              </li>
              <div
                className="btn-group"
              >
                <li>
                  <Link
                    replace={false}
                    to="/login"
                  >
                    <button
                      className="btn-success btn-sm"
                      type="button"
                    >
                      Login
                    </button>
                  </Link>
                </li>
                <li>
                  <Link
                    replace={false}
                    to="/register"
                  >
                    <button
                      className="btn-primary btn-sm"
                      type="button"
                    >
                      Register
                    </button>
                  </Link>
                </li>
              </div>
            </ul>
          </div>
        </nav>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="nav-wrapper "
          >
            <Link
              className="left"
              replace={false}
              style={
                Object {
                  "color": "white",
                  "fontSize": "30px",
                  "paddingLeft": "10px",
                  "textDecoration": "none",
                }
              }
              to="/"
            >
              The Events Manager
            </Link>
            <ul
              className="right hide-on-med-and-down"
              style={
                Object {
                  "paddingRight": "20px",
                }
              }
            >
              <li>
                <Link
                  replace={false}
                  to="/"
                >
                  Home
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/contact-us"
                >
                  Contact Us
                </Link>
              </li>
              <div
                className="btn-group"
              >
                <li>
                  <Link
                    replace={false}
                    to="/login"
                  >
                    <button
                      className="btn-success btn-sm"
                      type="button"
                    >
                      Login
                    </button>
                  </Link>
                </li>
                <li>
                  <Link
                    replace={false}
                    to="/register"
                  >
                    <button
                      className="btn-primary btn-sm"
                      type="button"
                    >
                      Register
                    </button>
                  </Link>
                </li>
              </div>
            </ul>
          </div>,
          "className": "indigo",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="left"
                replace={false}
                style={
                  Object {
                    "color": "white",
                    "fontSize": "30px",
                    "paddingLeft": "10px",
                    "textDecoration": "none",
                  }
                }
                to="/"
              >
                The Events Manager
              </Link>,
              <ul
                className="right hide-on-med-and-down"
                style={
                  Object {
                    "paddingRight": "20px",
                  }
                }
              >
                <li>
                  <Link
                    replace={false}
                    to="/"
                  >
                    Home
                  </Link>
                </li>
                <li>
                  <Link
                    replace={false}
                    to="/contact-us"
                  >
                    Contact Us
                  </Link>
                </li>
                <div
                  className="btn-group"
                >
                  <li>
                    <Link
                      replace={false}
                      to="/login"
                    >
                      <button
                        className="btn-success btn-sm"
                        type="button"
                      >
                        Login
                      </button>
                    </Link>
                  </li>
                  <li>
                    <Link
                      replace={false}
                      to="/register"
                    >
                      <button
                        className="btn-primary btn-sm"
                        type="button"
                      >
                        Register
                      </button>
                    </Link>
                  </li>
                </div>
              </ul>,
            ],
            "className": "nav-wrapper ",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "The Events Manager",
                "className": "left",
                "replace": false,
                "style": Object {
                  "color": "white",
                  "fontSize": "30px",
                  "paddingLeft": "10px",
                  "textDecoration": "none",
                },
                "to": "/",
              },
              "ref": null,
              "rendered": "The Events Manager",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li>
                    <Link
                      replace={false}
                      to="/"
                    >
                      Home
                    </Link>
                  </li>,
                  <li>
                    <Link
                      replace={false}
                      to="/contact-us"
                    >
                      Contact Us
                    </Link>
                  </li>,
                  undefined,
                  <div
                    className="btn-group"
                  >
                    <li>
                      <Link
                        replace={false}
                        to="/login"
                      >
                        <button
                          className="btn-success btn-sm"
                          type="button"
                        >
                          Login
                        </button>
                      </Link>
                    </li>
                    <li>
                      <Link
                        replace={false}
                        to="/register"
                      >
                        <button
                          className="btn-primary btn-sm"
                          type="button"
                        >
                          Register
                        </button>
                      </Link>
                    </li>
                  </div>,
                ],
                "className": "right hide-on-med-and-down",
                "style": Object {
                  "paddingRight": "20px",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      replace={false}
                      to="/"
                    >
                      Home
                    </Link>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Home",
                      "replace": false,
                      "to": "/",
                    },
                    "ref": null,
                    "rendered": "Home",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      replace={false}
                      to="/contact-us"
                    >
                      Contact Us
                    </Link>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Contact Us",
                      "replace": false,
                      "to": "/contact-us",
                    },
                    "ref": null,
                    "rendered": "Contact Us",
                    "type": [Function],
                  },
                  "type": "li",
                },
                undefined,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <li>
                        <Link
                          replace={false}
                          to="/login"
                        >
                          <button
                            className="btn-success btn-sm"
                            type="button"
                          >
                            Login
                          </button>
                        </Link>
                      </li>,
                      <li>
                        <Link
                          replace={false}
                          to="/register"
                        >
                          <button
                            className="btn-primary btn-sm"
                            type="button"
                          >
                            Register
                          </button>
                        </Link>
                      </li>,
                    ],
                    "className": "btn-group",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <Link
                          replace={false}
                          to="/login"
                        >
                          <button
                            className="btn-success btn-sm"
                            type="button"
                          >
                            Login
                          </button>
                        </Link>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": <button
                            className="btn-success btn-sm"
                            type="button"
                          >
                            Login
                          </button>,
                          "replace": false,
                          "to": "/login",
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "Login",
                            "className": "btn-success btn-sm",
                            "type": "button",
                          },
                          "ref": null,
                          "rendered": "Login",
                          "type": "button",
                        },
                        "type": [Function],
                      },
                      "type": "li",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <Link
                          replace={false}
                          to="/register"
                        >
                          <button
                            className="btn-primary btn-sm"
                            type="button"
                          >
                            Register
                          </button>
                        </Link>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": <button
                            className="btn-primary btn-sm"
                            type="button"
                          >
                            Register
                          </button>,
                          "replace": false,
                          "to": "/register",
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "Register",
                            "className": "btn-primary btn-sm",
                            "type": "button",
                          },
                          "ref": null,
                          "rendered": "Register",
                          "type": "button",
                        },
                        "type": [Function],
                      },
                      "type": "li",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "ul",
            },
          ],
          "type": "div",
        },
        "type": "nav",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
